graspologic/plot/plot.py

```diff
@@ -673,6 +677,240 @@ def pairplot(
+    means : array-like, shape (n_components, n_features)
+        Estimated means from gmm
+    covariances : array-like, shape (with 'full') (n_components, n_features, n_features)
+        estimated covariances from gmm
+    ax : axis object
+        The location where plot_ellipse will plot
+    j : int
+        column index of feature of interest from X which will be the x-axis data.
+    k : int
+        column index of feature of interest from X which will be the y-axis data.
+    lab_dict : dict
+        Dictionary of labels mapped to unique number
+    ind_to_lab : dict
+        The reverse of lab_dict
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    palette : str, dict, optional, default: 'Set1'
+    """
+    sns.color_palette(palette, as_cmap=True)
+    for i, (mean, covar) in enumerate(zip(means, covariances)):
+        v, w = linalg.eigh(covar)
+        v = 2.0 * np.sqrt(2.0) * np.sqrt(v)
+        u = w[0] / linalg.norm(w[0])
+        if lab_dict and ind_to_lab:
+            # Plot the scatter
+            sns.scatterplot(X[Y_ == i, j], X[Y_ == i, k], ax=ax, label=ind_to_lab[i])
+            # Plot an ellipse to show the Gaussian component
+            angle = np.arctan(u[1] / u[0])
+            angle = 180.0 * angle / np.pi
+            ell = mpl.patches.Ellipse(
+                [mean[j], mean[k]], v[0], v[1], 180.0 + angle, color="red"
+            )
+        else:
+            if not np.any(Y_ == i):
+                continue
+            sns.scatterplot(
+                X[Y_ == i, j], X[Y_ == i, k], ax=ax, label="Block {}".format(i)
+            )
+            # Plot an ellipse to show the Gaussian component
+            angle = np.arctan(u[1] / u[0])
+            angle = 180.0 * angle / np.pi
+            ell = mpl.patches.Ellipse(
+                [mean[j], mean[k]], v[0], v[1], 180.0 + angle, color="red"
+            )
+        ell.set_clip_box(ax.bbox)
+        ell.set_alpha(alpha)
+        ax.add_artist(ell)
+        ax.set_xticks([]), ax.set_yticks([])
+        ax.set_ylabel(k), ax.set_xlabel(k)
+
+
+def pairplot_with_gmm(
+    X,
+    gmm,
+    labels=None,
+    title=None,
+    context="talk",
+    font_scale=1,
+    alpha=0.7,
+    palette="Set1",
+):
+    r"""
+    Plot pairwise relationships in a dataset.
+
+    By default, this function will create a grid of Axes such that each dimension
+    in data will by shared in the y-axis across a single row and in the x-axis
+    across a single column.
+
+    The off-diagonal Axes show the pairwise relationships displayed as scatterplot.
+    Using the inputted gmm an ellipse to show the Gaussian component will be plotted.
+    The diagonal Axes show the univariate distribution of the data for that
+    dimension displayed as kernel density estimates (KDEs).
+
+    Read more in the :ref:`tutorials <plot_tutorials>`
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        Input data.
```

Comment:
@bdpedigo I am also confused as to why i am getting " /home/runner/work/graspologic/graspologic/graspologic/plot/plot.py:docstring of graspologic.plot.pairplot_with_gmm:17:Unexpected indentation." from the build-documentation checks. I ran with black and nothing changed... is there something else I can do?